src/main/java/net/snowflake/client/core/SessionUtil.java

```diff
@@ -1154,26 +1154,29 @@ private static String federatedFlowStep4(
               loginInput.getHttpClientSettingsKey());
 
       // step 5
-      String postBackUrl = getPostBackUrlFromHTML(responseHtml);
-      if (!isPrefixEqual(postBackUrl, loginInput.getServerUrl())) {
-        URL idpDestinationUrl = new URL(postBackUrl);
-        URL clientDestinationUrl = new URL(loginInput.getServerUrl());
-        String idpDestinationHostName = idpDestinationUrl.getHost();
-        String clientDestinationHostName = clientDestinationUrl.getHost();
-
-        logger.error(
-            "The Snowflake hostname specified in the client connection {} does not match "
-                + "the destination hostname in the SAML response returned by the IdP: {}",
-            clientDestinationHostName,
-            idpDestinationHostName);
-
-        // Session is in process of getting created, so exception constructor takes in null session
-        // value
-        throw new SnowflakeSQLLoggedException(
-            null,
-            ErrorCode.IDP_INCORRECT_DESTINATION.getMessageCode(),
-            SqlState.SQLCLIENT_UNABLE_TO_ESTABLISH_SQLCONNECTION
-            /* session = */ );
+      if (!loginInput.getDisableSamlUrlCheck()) {
+        String postBackUrl = getPostBackUrlFromHTML(responseHtml);
+        if (!isPrefixEqual(postBackUrl, loginInput.getServerUrl())) {
+          URL idpDestinationUrl = new URL(postBackUrl);
+          URL clientDestinationUrl = new URL(loginInput.getServerUrl());
+          String idpDestinationHostName = idpDestinationUrl.getHost();
+          String clientDestinationHostName = clientDestinationUrl.getHost();
+
+          logger.error(
+              "The Snowflake hostname specified in the client connection {} does not match "
+                  + "the destination hostname in the SAML response returned by the IdP: {}",
+              clientDestinationHostName,
+              idpDestinationHostName);
+
+          // Session is in process of getting created, so exception constructor takes in null
+          // session
+          // value
+          throw new SnowflakeSQLLoggedException(
+              null,
+              ErrorCode.IDP_INCORRECT_DESTINATION.getMessageCode(),
+              SqlState.SQLCLIENT_UNABLE_TO_ESTABLISH_SQLCONNECTION
+              /* session = */ );
```

Comment:
we can remove this comment (or move it to the line where null is set)

Reply:
Removed.